hipreport.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+
+# openconnect will call this script with the follow command-line
+# arguments, which are needed to populate the contents of the
+# HIP report:
+#
+#   --cookie: a URL-encoded string, as output by openconnect
+#             --authenticate --protocol=gp, which includes parameters
+#             from the /ssl-vpn/login.esp response
+#
+#   --client-ip: IPv4 address allocated by the GlobalProtect VPN for
+#                this client (included in /ssl-vpn/getconfig.esp
+#                response)
+#
+#   --md5: The md5 digest to encode into this HIP report. I'm not sure
+#          exactly what this is the md5 digest *of*, but all that
+#          really matters is that the value in the HIP report
+#          submission should match the value in the HIP report check.
+
+# Read command line arguments into variables
+COOKIE=
+IP=
+MD5=
+
+while [ "$1" ]; do
+    if [ "$1" = "--cookie" ];    then shift; COOKIE="$1"; fi
```

Comment:
Can we re-write this in case style?  It's more efficient, idiomatic, and readable.

So (untested, watch for typos!):
```bash
while [[ $# -gt 0 ]]
do
  arg="$1"
  case $arg in
    --cookie)
      COOKIE="$2"
      shift 2
      ;;
    --client-ip)
      IP="$2"
      shift 2
      ;;
    ...
    *)
      >&2 echo "ERROR: Unknown argument $arg"
  esac
done
```